server-test/src/test/scala/testpkg/BuildServerTest.scala

```diff
@@ -94,6 +94,35 @@ object BuildServerTest extends AbstractServerTest {
     assert(res.statusCode == StatusCode.Success)
   }
 
+  test("buildTarget/compile - reports compilation progress") { _ =>
+    val buildTarget = buildTargetUri("runAndTest", "Compile")
+    svr.sendJsonRpc(
+      s"""{ "jsonrpc": "2.0", "id": "33", "method": "buildTarget/compile", "params": {
+         |  "targets": [{ "uri": "$buildTarget" }]
+         |} }""".stripMargin
+    )
+
+    assert(svr.waitForString(10.seconds) { s =>
+      s.contains("build/taskStart") &&
+      s.contains(""""message":"Compiling runAndTest"""")
+    })
+
+    assert(svr.waitForString(10.seconds) { s =>
+      s.contains("build/taskProgress") &&
+      s.contains(""""message":"Compiling runAndTest (15%)"""")
+    })
+
+    assert(svr.waitForString(10.seconds) { s =>
+      s.contains("build/taskProgress") &&
+      s.contains(""""message":"Compiling runAndTest (100%)"""")
+    })
+
+    assert(svr.waitForString(10.seconds) { s =>
+      s.contains("build/taskFinish") &&
+      s.contains(""""message":"Compiled runAndTest"""")
+    })
```

Comment:
I'm not sure which one of the `waitForString`, but these seem to be flaky at least directly after when Scala version was bumped to Scala 2.12.15. Like, sometimes the test would pass and other times it doesn't. I'm going to try to bump the timeout to 60s and see if that would help.

Reply:
Does it work better after increasing the timeout?

If not then it probably means that test progress is not deterministic, depends on the timing.